.ci_support/osx_64_.yaml

```diff
@@ -50,20 +52,18 @@ macos_machine:
 - x86_64-apple-darwin13.4.0
 minuit2:
 - '6'
-mpich:
-- '4'
 ndarray:
 - '1'
 numpy:
-- '1.23'
+- '1.24'
 pin_run_as_build:
   boost:
     max_pin: x.x.x
   python:
     min_pin: x.x
     max_pin: x.x
 python:
-- 3.10.* *_cpython
+- 3.11.* *_cpython
 python_impl:
 - cpython
 starlink_ast:
```

Comment:
What's the purpose of the version number below?  This is not the current version but I don't know whether it has to be.

Reply:
This is the actual version used.  It's pinned here (via `conda_build_config.yaml`) for binary compatibility.

Reply:
In other words, if we should move to a newer version, please say so.

Reply:
Right, this would be the time to look through all these pinned versions, right?

Reply:
Okay. We may as well go to 9.2.10 then if we are starting an entirely new env with new python version.

Reply:
Okay, looking through the ones that are in https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/main/recipe/conda_build_config.yaml I see that (a) we should update our gfortran to 12 (this also allows long double REAL*16 support on osx-arm64 in case anybody needs that; (b) the starlink_ast global pin is still at 9.2.7.  So I think we should stick with that, but maybe we want to poke the system to get this upgraded globally at some point.  (I don't know how to do that).

Everything else looks fine to my eye.

Reply:
I don't understand how I can ever update starlink-ast again if it's a global pin.

Reply:
You can install it in your own env, but you won't be able to install any conda forge packages that depend on it because of the ABI mismatch.  So you have to do some sort of PR to trigger a global migration to the new version.

Reply:
I will give you a (virtual) dollar if any conda package depends on it (not including us). Is this all because semantic versioning was not promised?

Reply:
github search says that you're right.  I think then we can/should upgrade it, but probably also do a PR to the global thingy.

Reply:
Looks like we can't do gfortran 12 without gcc 12.

Reply:
Is that a linux thing or everything?

Reply:
Also, the global pin for gcc is also 12: https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/main/recipe/conda_build_config.yaml#L7. I think we should be 12 across the board.

Reply:
(Sorry I didn't look at this before, I thought we had updated all of those...)